geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreFactory.scala

```diff
@@ -183,8 +217,35 @@ object AccumuloDataStoreFactory {
     security.getAuthorizationsProvider(params, auths)
   }
 
-  def canProcess(params: JMap[String,Serializable]): Boolean =
-    params.containsKey(instanceIdParam.key) || params.containsKey(connParam.key)
+  // Kerberos is only available in Accumulo >= 1.7.
+  // Probe for whether part of the API is available
+  // Note: doesn't confirm whether correctly configured for Kerberos e.g. core-site.xml on CLASSPATH
+  def isKerberosAvailable() = {
+    try {
+      val conf = new org.apache.accumulo.core.client.ClientConfiguration()
+      val withSasl = conf.getClass().getMethod("withSasl", classOf[Boolean])
+      true
+    }
+    catch {
+      case nsme: NoSuchMethodException => false
+    }
+  }
+
+  def canProcess(params: JMap[String,Serializable]): Boolean = {
+
+    val instanceIdOrConnection = params.containsKey(instanceIdParam.key) || params.containsKey(connParam.key)
+
+    // Is valid password set?
+    val hasPassword = params.containsKey(passwordParam.key) && passwordParam.lookup[String](params) != null
+
+    // Is valid keytabPath set?
+    val hasKeytabPath = params.containsKey(keytabPathParam.key) && keytabPathParam.lookup[String](params) != null
+
+    instanceIdOrConnection && (
+        (!hasPassword && !hasKeytabPath) || // neither set - included for backwards compatibility, if connector provided
```

Comment:
could we split out the cases where instanceId is provided vs connector? If connector is provided, we don't need anything else, but otherwise we would need  user + credentials